### ISSUE TITLE
Is there a centralized place to access response Context-Length for telemetry

### ISSUE THREAD

**Issue by AlexDenton** [dotnet/aspnetcore#33755]
I want to add telemetry either as a log or to Application insights for the Content-Length of a response in dotnet core 2.2. I've tried a number of places in middleware and different filters in the request pipeline. No where that I've checked has the Content-Length materialized--it's always null.

My alternate solution was to check the response stream myself and compute the length but I'd really prefer not to re-read the stream if I don't have to. Is there somewhere in the dotnet core request pipeline that I can hook into for that information?

**Comment by Tratcher** [comment]
Content-Length is the actual response header sent to the client, it is only set by the response generator (e.g. the application), not the framework.

The only place that tracks actual bytes written is the server. I'll have to check of that data is available somewhere.

One way to track this would be to wrap the response stream and count the bytes as they're written.

FYI 2.2 is out of support.

**Comment by AlexDenton** [comment]
@Tratcher Fair enough. Counting the bytes on the way out is pretty clever. Might give that a go. 

Seems like this would a be a common need from a telemetry perspective. Definitely let me know if you find a better/more official way to do this!

**Comment by BrennanConroy** [comment]
Triage: no way (other than wrapping the stream) to do this today.
Would be interesting as a potential future enhancement.
